### PR TITLE
WIP: Add unix sockets support

### PR DESCRIPTION
**Pull request by Rush**
Ref #249

### CODE REVIEW

**Comment by Rush** [comment]
Looked easy enough. Nice codebase. Sadly I cannot test Windows build ... do you have a CI set up?

Couple of remarks:
* No `.gitignore`
* Hard coded paths https://github.com/uWebSockets/uWebSockets/blob/master/main.cpp#L17
* Would be nice if CMake had a target to also build the Node addon.

File: src/Server.h [hunk 1]
```diff
@@ -7,6 +7,7 @@
 #include <functional>
 #include <uv.h>
 #include <openssl/ossl_typ.h>
+#include <sys/un.h>
```
Comment (Rush) [inline src/Server.h]: This needs an `#ifdef` @alexhultman - maybe checking if it's not Windows is enough?

**Comment by alexhultman** [comment]
It is a very elegant interface to pass a port number or a string. I like that and it will certainly be merged at some point. I'm a little busy currently, working on the big rewrite.

**Comment by alexhultman** [comment]
This will need to be updated for master

**Comment by Rush** [comment]
Kinda busy at this moment. Will look into it in a few days. Thanks.

**Comment by Rush** [comment]
I see there's a new release. Could you merge it to master?

**Comment by alexhultman** [comment]
@Rush How do you mean?

**Comment by Rush** [comment]
Hey Alex, it's very unfair comment you posted here https://github.com/uWebSockets/uWebSockets/issues/249#issuecomment-264666228

You must have mistook me with somebody else. I never promised any Travis setup.

The only thing I did was requesting UNIX socket support which I did implement in this PR here. Then you said you're not gonna merge it cause you were re-writing everything.  I thought I would have time to work on it but I didn't. It doesn't mean I won't work on it in the future.

And my view is, since you were re-writing everything you could have included my PR before your re-write, but well...

Also, please remove your false comment about me promising Travis support and please re-open this PR.

**Pull request closed without merging**